### PR TITLE
Optimize review page slightly by grouping and removing some queries

### LINKED ISSUE #16096: Optimize calls to version.is_ready_for_auto_approval property and addon.find_latest_version() in reviewer tools
There are 2 obvious sets of duplicate queries in reviewer tools at the moment:

- `<Version>.is_ready_for_auto_approval` is a property we only use in a couple places (in reviewer tools an devhub) to show/avoid showing some extra info to reviewers / developers about a version depending on whether it's auto approvable or not.
It's implemented in a DRY way, it calls `Version.objects.auto_approve()` under the hood, restricting with the `pk` of the version we're looking at. Unfortunately, this means it's causing a database query for every version it's called. This matters in reviewer tools where we display 10 versions on each page.
We should optimize this, maybe making this a `@cached_property` that we could fill for all instances in a queryset in one go, or by finding a way to still share the code with the manager method while using the `Addon` and `File` instances that are already loaded.
- `files_view.html`, which is rendered for each version through `all_distinct_files()` template tag, does `{% set latest_version_in_channel=addon.find_latest_version(channel=version.channel) %}` repeating the two queries that method executes for every version. It should be set before in context, but we should be careful since it's a template tag : nothing else is using it at the moment, but we may want to explicitly pass the `latest_version` to the tag so that it doesn't fail suddenly if we change the context or use the tag in a different manner in the future.

Doing ~ 30 fewer queries per page seems like a nice perf win...

### PR DESCRIPTION
**Pull request by diox**
Around 39 fewer queries for an add-on with 10 versions being displayed.

Fixes #16096

### CODE REVIEW

File: src/olympia/versions/models.py [hunk 1]
```diff
@@ -675,6 +675,22 @@ def transformer_license(cls, versions):
...
+        ids = set(v.id for v in versions)
+        if not ids:
+            return
+
+        auto_approvable = dict(
+            Version.objects.auto_approvable()
+                   .annotate(
+                       is_ready_for_auto_approval=Value(True, BooleanField()))
+                   .values_list('pk', 'is_ready_for_auto_approval')
+        )
```
Comment (eviljeff) [inline src/olympia/versions/models.py:689]: So, this is just a dict of `pk: True`?  Why not go with the more simple solution of getting a list of pks and setting true if the version pk is in that list?

File: src/olympia/reviewers/tests/test_views.py [hunk 2]
```diff
@@ -3340,7 +3340,76 @@ def test_item_history_pagination(self):
             # Add versions 1.0 to 1.9
             version_factory(
                 addon=addon, version=f'1.{i}', created=self.days_ago(365 - i))
-        response = self.client.get(self.url)
+        with self.assertNumQueries(60):
```
Comment (eviljeff) [inline src/olympia/reviewers/tests/test_views.py:3343]: 🐼

**Review by eviljeff — APPROVED** [review]
it's something!

File: src/olympia/versions/models.py (hunk 1 above)
Comment (diox) [reply src/olympia/versions/models.py:689]: Yeah, also, I should filter by `pk__in=ids` ... I'll change this.

Comment (diox) [reply src/olympia/versions/models.py]: https://github.com/mozilla/addons-server/pull/16105/commits/5263dc2c5789318325bb4b5fce314a09bc21a3bd

**Pull request merged**